src/stack/Stack.java

```diff
@@ -19,7 +19,11 @@ public Stack(int size) {
 	 * Add element into stack
 	 */
 	public void push(int value) {
-
+		if (this.tos == data.length) {
```

Comment:
1. You have to type data.length -1 here, because if you don't put it, it will trow IndexAuthOfBoundException
2. then you will call your ensureCapacity() method